packages/create-react-app/createReactApp.js

```diff
@@ -545,9 +544,12 @@ function checkIfOnline(useYarn) {
     return Promise.resolve(true);
   }
 
-  return new Promise(resolve => {
-    dns.lookup('registry.yarnpkg.com', err => {
-      resolve(err === null);
-    });
+  return new Promise((resolve, reject) => {
+    const yarnRegistry = execSync('yarnpkg config get registry').toString().trim();
+    const registryStream = hyperquest(yarnRegistry);
+
+    registryStream.on('error', reject);
```

Comment:
I don't think we use rejection like this. We actually expect Promise to resolve with `false` when you are offline (see the calling code).